### DISCUSSION TITLE
Show rule to replace multiple tokens

### DISCUSSION THREAD

**Discussion by davystrong** [typst/typst#4734] [Q&A] [answered]
Hello!

I'm trying to do some slightly more advanced text replacement. The [docs](https://typst.app/docs/reference/styling/#show-rules) show an example of using show rules to replace text:

```
#show "badly": "great"
```

This works fine with a single token (/element/not sure what it's called), but what if I want to replace multiple tokens with one new one? For example, if I want to replace the `"text"` with `something else`. Because `"text"` gets parsed as `([smartquote()],[text],[smartquote()])` (i.e. 3 tokens) I can't do this directly with a normal show rule.

I tried to do this with a bare show rule, manually iterating through the tokens. This is fine, but it fails when an element, such as `block`, is added which has children. I can, of course, manually handle blocks and iterate through their children, but then the same thing happens with styled blocks, and that has to be handled differently, and so on.

I also tried making use of context and specific show rules for `smartquote` and `text` strings. I can quite easily identify the pattern and add the new text, but because at that point the "cursor" has moved past original text, I can't remove it. In other words, it's quite easy to get `"text"something else`, but removing the `"text"` is seemingly impossible.

Can anyone help me? I feel I may be missing some neat way of doing this!

**Comment by laurmaedje** [comment, marked as answer]
I'm afraid there is no easy way. It's simply a compiler bug: #86

**Reply by davystrong** [reply]
Ok, thanks! In my case they're not all text nodes, but I guess it still applies.

**Reply by laurmaedje** [reply]
Even the smart quotes would ideally be converted to their string equivalents by Typst.
